extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueBridgeNupnpDiscovery.java

```diff
@@ -0,0 +1,141 @@
+    protected void startScan() {
+        discoverHueBridges();
+    }
+
+    /**
+     * Discover available Hue Bridges and then add them in the discovery inbox
+     */
+    private void discoverHueBridges() {
+        // get bridge list
+        List<BridgeJsonParameters> bridgeList = getBridgeList();
+        if (bridgeList != null) {
+            for (BridgeJsonParameters bridge : bridgeList) {
+                if (bridge.getId() != null && bridge.getInternalipaddress() != null) {
+                    // check if bridge is reachable, and if 'id' contains 'fffe' at the middle
+                    if (isHueBridge(bridge.getInternalipaddress()) && bridge.getId().substring(6, 10).equals("fffe")) {
+                        // 'fffe' should be removed from 'id', to match with 'id' in description.xml file
+                        ThingUID uid = new ThingUID(THING_TYPE_BRIDGE,
+                                bridge.getId().substring(0, 6) + bridge.getId().substring(10));
+                        if (uid != null) {
+                            Map<String, Object> properties = new HashMap<>(2);
+                            properties.put(HOST, bridge.getInternalipaddress());
+                            properties.put(SERIAL_NUMBER, bridge.getId().replace("fffe", ""));
+                            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                                    .withLabel("Philips hue (" + bridge.getInternalipaddress() + ")")
+                                    .withRepresentationProperty(SERIAL_NUMBER).build();
+                            thingDiscovered(result);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * To check if Bridge is reachable and if it is Hue Bridge.
+     *
+     * @param host (Bridge IP, must not be null)
+     * @return true if Bridge is reachable and is Hue, otherwise false
+     */
+    private boolean isHueBridge(String host) {
+        try {
+            String url = "http://" + host + "/description.xml";
+            String description = HttpUtil.executeUrl("GET", url, timeout);
+            return description.contains("<modelName>Philips hue");
+        } catch (IOException e) {
+            logger.debug("Failure accessing description.xml file of Hue Bridge with IP address - {}.", host);
+        }
+        return false;
+    }
+
+    /**
+     * Use the Philips Hue NUPnP service to find Hue Bridges in local Network.
+     *
+     * @return a list of available Hue Bridges
+     */
+    private List<BridgeJsonParameters> getBridgeList() {
+        try {
+            Gson gson = new Gson();
+            String json = HttpUtil.executeUrl("GET", url, timeout);
+            return gson.fromJson(json, new TypeToken<List<BridgeJsonParameters>>() {
+            }.getType());
+        } catch (IOException e) {
+            logger.error("Error accessing the Philips Hue NUPnP service.", e);
```

Comment:
Which is perfectly valid, if there is no Internet access available. Hence this must not result in any errors being logged.